code-experiments/src/c_linear.c

```diff
@@ -310,22 +317,31 @@ static coco_problem_t *c_linear_single_cons_bbob_problem_allocate(const size_t f
                                 + 50000 * constraint_number);
   srand(seed_cons_i);
   
+  /* The constraints gradients are scaled with random numbers
+   * 10**U[0,1] and 10**U_i[0,2], where U[a, b] is uniform in [a,b] 
+   * and only U_i is drawn for each constraint individually. 
+   * The random number 10**U[0,1] is given by the variable 'factor1' 
+   * while the random number 10**U_i[0,2] is calculated below and 
+   * stored as 'factor2'. (The exception is when the number of
+   * constraints is n+1, in which case 'factor2' defines a random
+   * number 10**U_i[0,1])
+   */
+     
+  if (number_of_linear_constraints == dimension + 1)
+    exp2 = rand() / (RAND_MAX / (1 + 1) + 1);
```

Comment:
I had problems using coco_random_normal() to generate the gradients a long time ago. I can't quite recall what happened, but, after having generated a certain number of pseudorandom numbers, a segmentation fault occurred. The lack of comments in the functions in coco_random.c makes the debugging and its use rather difficult. Of course, anyone that knows coco_random.c well can change c_linear.c in order to use its functions.